library/src/main/java/com/datatorrent/lib/db/jdbc/AbstractJdbcPollInputOperator.java

```diff
@@ -136,13 +139,24 @@
   protected transient PreparedStatement ps;
   protected boolean isPollerPartition;
 
-  protected transient MutablePair<Integer, Integer> currentWindowRecoveryState;
-
   private transient int lastOffset;
+  private transient Object prevKey;
+  private transient Object lastKey;
+
+  /**
+   * The candidate key/offset pair identified by the fetch thread
```

Comment:
Suggest using "poller thread"  instead of "fetch thread" since the thread class is called "DBPoller", reader will be able correlate the two more easily.